File: javascript/core/src/verbs/rename.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { TableStep } from './nodeFactories/index.js'
import type { InputColumnRecordArgs } from './types.js'
import { stepNodeFactory } from './nodeFactories/StepNode.js'

export type RenameArgs = InputColumnRecordArgs

export const renameStep: TableStep<RenameArgs> = (input, { columns }) =>
	input.rename(columns)

export const rename = stepNodeFactory(renameStep)

File: python/datashaper/datashaper/verbs/rename.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Rename verb implementation."""

import pandas as pd
from reactivedataflow import ConfigPort, InputPort, verb

from datashaper import DEFAULT_INPUT_NAME

from .decorators import OutputMode, copy_input_tables, wrap_verb_result


@verb(
    name="rename",
    ports=[
        InputPort(name=DEFAULT_INPUT_NAME, parameter="table", required=True),
        ConfigPort(name="columns", required=True),
    ],
    adapters=[
        copy_input_tables("table"),
        wrap_verb_result(mode=OutputMode.Table),
    ],
)
def rename(table: pd.DataFrame, columns: dict[str, str]) -> pd.DataFrame:
    """Rename verb implementation."""
    return table.rename(columns=columns)
